api/src/pcapi/core/finance/api.py

```diff
@@ -995,43 +998,39 @@ def _mark_as_processed(pricings: sqla_orm.Query) -> None:
                         .all()
                     )
 
-                    if total > 0:
-                        override_incident_debit_note = any(
-                            incident.forceDebitNote for incident in all_current_incidents
+                    override_incident_debit_note = any(incident.forceDebitNote for incident in all_current_incidents)
+                    # Last cashflow where we effectively paid (successfully or not) the pro
+                    last_cashflow = (
+                        models.Cashflow.query.filter(
+                            models.Cashflow.reimbursementPointId == reimbursement_point_id,
+                            models.Cashflow.status == models.CashflowStatus.ACCEPTED,
                         )
-                        # Last cashflow where we effectively paid (successfully or not) the pro
-                        last_cashflow = (
-                            models.Cashflow.query.filter(
-                                models.Cashflow.reimbursementPointId == reimbursement_point_id,
-                                models.Cashflow.status == models.CashflowStatus.ACCEPTED,
+                        .order_by(models.Cashflow.creationDate.desc())
+                        .first()
+                    )
+                    if last_cashflow:
+                        last_cashflow_age = (datetime.datetime.utcnow() - last_cashflow.creationDate).days
```

Comment:
[détail / style] On a `last_cashflow_age = 0` environ 30 lignes plus haut. Je trouve qu'il serait plus lisible que ce soit défini ici (proche de l'endroit où on le définit avec une autre valeur).


```suggestion
                    if last_cashflow:
                        last_cashflow_age = (datetime.datetime.utcnow() - last_cashflow.creationDate).days
                    else:
                        last_cashflow_age = 0
```

Idem dans `_generate_cashflows()`.